highline.rb

```diff
@@ -73,14 +73,14 @@
 ]
 
 all_jobs.each do |job|
-  keywords.each do |included_words, excluded_words|
-    passes = job.include?(included_words)
-    fails = job.include?(excluded_words)
+  keywords.each do |hash|
+    passes = hash["include"].any? { |word| job.include?(word) }
+    fails  = hash["exclude"].any? { |word| job.include?(word) }
     if passes && !fails #contains any of the included words and none of the excluded words]
       relevant_jobs.push(job)
     end
-  end 
-end 

```

Comment:
Each item in 'keywords' is a hash, so we have to reference that first, and then get the includes words via hash["include"] and hash["exclude"]. Also .include?, when called on a string, has to take a string, so we have to go through the "include" or "exclude" arrays, using the shortcut [.any?](https://ruby-doc.org/core-2.2.0/Enumerable.html#method-i-any-3F), and ask if anything in the those arrays is included in the 'job' string.

Reply:
Okay, so "keywords" is an array containing hashes which then contain arrays... 

That is one of the trickiest things for me to get straight.  :/